cmd/primateRecon/primateRecon.go

```diff
@@ -111,30 +83,47 @@ func hcaIsPresent(human, bonobo, chimp, gorilla, organutan dna.Base) bool {
 	return false
 }
 
+func hgaIsPresent(human, bonobo, chimp, gorilla, orangutan dna.Base) bool {
+	if baseIsPresent(gorilla) && (baseIsPresent(human) || baseIsPresent(chimp) || baseIsPresent(bonobo)) {
+		return true
+	}
+	if baseIsPresent(orangutan) && (baseIsPresent(gorilla) || baseIsPresent(human) || baseIsPresent(chimp) || baseIsPresent(bonobo)) {
+		return true
+	}
+	return false
+}
+
 func reconHcaBase(root, humanNode, chimpNode, nodeToRecon *expandedTree.ETree, position int, probThreshold float64, nonBiasProbThreshold float64, humanBias bool, chimpBias bool) {
 	var likelihoods []float64
 	var nextBase dna.Base
 	reconstruct.SetState(root, position)
 	likelihoods = reconstruct.FixFc(root, nodeToRecon)
 	if humanBias {
-		nextBase = likelihoodsToBaseHumanBias(likelihoods, humanNode.Fasta.Seq[position], probThreshold, nonBiasProbThreshold)
+		nextBase = likelihoodsToBaseBias(likelihoods, humanNode.Fasta.Seq[position], probThreshold, nonBiasProbThreshold)
 	} else if chimpBias {
-		nextBase = likelihoodsToBaseChimpBias(likelihoods, chimpNode.Fasta.Seq[position], probThreshold, nonBiasProbThreshold)
+		nextBase = likelihoodsToBaseBias(likelihoods, chimpNode.Fasta.Seq[position], probThreshold, nonBiasProbThreshold)
 	} else {
 		nextBase = likelihoodsToBaseUnbiased(likelihoods, probThreshold)
 	}
 	nodeToRecon.Fasta.Seq = append(nodeToRecon.Fasta.Seq, nextBase)
 }
 
-func primateReconMle(inFastaFilename string, inTreeFilename string, humanBias bool, chimpBias bool, probThreshold float64, nonHumanProbThreshold float64, outputFastaFilename string) {
+func reconHgaBase(root, gorillaNode, nodeToRecon *expandedTree.ETree, position int, probThreshold float64, nonBiasProbThreshold float64) {
+	var likelihoods []float64
+	var nextBase dna.Base
+	reconstruct.SetState(root, position)
+	likelihoods = reconstruct.FixFc(root, nodeToRecon)
+	nextBase = likelihoodsToBaseBias(likelihoods, gorillaNode.Fasta.Seq[position], probThreshold, nonBiasProbThreshold)
+	nodeToRecon.Fasta.Seq = append(nodeToRecon.Fasta.Seq, nextBase)
+}
+
+func primateReconHcaMle(inFastaFilename string, inTreeFilename string, humanBias bool, chimpBias bool, probThreshold float64, nonHumanProbThreshold float64, outputFastaFilename string) {
```

Comment:
I noticed that primateReconHcaMle and primateReconHgaMle have different numbers of input variables, since the Hga version does not have humanBias bool or chimpBias bool. I wonder if having the same number of input variables will be easier to write in the program, in case you expand to other types of biased common ancestor reconstruction? This way, maybe you won't have to write as many checks in the main function to determine if the input variables are legal and point them to the correct primateReconMle function. This is just an idea since your current style already works

Reply:
yes we would want that if we ever decided to make a human-biased HGA estimate. For now, I only wrote the options I needed for the tasks at hand